provider/marathon/marathon_test.go

```diff
@@ -934,440 +699,563 @@ func TestMarathonGetProtocol(t *testing.T) {
 		c := c
 		t.Run(c.desc, func(t *testing.T) {
 			t.Parallel()
+			provider := &Provider{}
 			actual := provider.getProtocol(c.application)
 			if actual != c.expected {
-				t.Errorf("got protocol '%s', want '%s'", actual, c.expected)
+				t.Errorf("actual %q, expected %q", actual, c.expected)
 			}
 		})
 	}
 }
 
-func TestMarathonGetPassHostHeader(t *testing.T) {
-	provider := &Provider{}
-
-	applications := []struct {
+func TestMarathonGetSticky(t *testing.T) {
+	cases := []struct {
+		desc        string
 		application marathon.Application
 		expected    string
 	}{
 		{
-			application: marathon.Application{
-				Labels: &map[string]string{}},
+			desc:        "label missing",
+			application: application(),
```

Comment:
In general, I'd prefer not to use longer names in order to keep things concise since builder functions are used very frequently. IMHO, we shouldn't have extra verbs on builder operations.

`anApplication` looks quite weird to my eyes. :) How about `createApp` and `createTask` instead?